go/common/crypto/tls/cert.go

```diff
@@ -52,13 +53,7 @@ func LoadOrGenerate(certPath, keyPath, commonName string) (*tls.Certificate, err
 	return cert, nil
 }
 
-// Generate generates a new TLS certificate.
-func Generate(commonName string) (*tls.Certificate, error) {
-	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
-	if err != nil {
-		return nil, fmt.Errorf("tls: failed to generate keypair: %w", err)
-	}
-
+func makeCertificate(commonName string, pubKey crypto.PublicKey, privKey ed25519.PrivateKey) (*tls.Certificate, error) {
```

Comment:
Why do you need both public and private keys here when you can do `privKey.Public()` to get the public part?